packages/medusa/src/services/__tests__/product-category.ts

```diff
@@ -46,4 +46,73 @@ describe("ProductCategoryService", () => {
       )
     })
   })
+
+  describe("delete", () => {
+    const productCategoryRepository = MockRepository({
+      findOne: query => {
+        if (query.where.id === "not-found") {
+          return Promise.resolve(undefined)
+        }
+
+        if (query.where.id === "with-children") {
+          return Promise.resolve({
+            id: IdMap.getId("with-children"),
+            category_children: [{
+              id: IdMap.getId("skinny-jeans"),
+            }]
+          })
+        }
+
+        return Promise.resolve({
+          id: IdMap.getId("jeans"),
+          category_children: []
+        })
+      },
+      findDescendantsTree: (productCategory) => {
+        return Promise.resolve(productCategory)
+      },
+      softRemove: (productCategory) => {
+        return Promise.resolve(productCategory)
+      }
+    })
+
+    const productCategoryService = new ProductCategoryService({
+      manager: MockManager,
+      productCategoryRepository,
+    })
+
+    beforeEach(async () => { jest.clearAllMocks() })
+
+    it("successfully deletes a product category", async () => {
+      const result = await productCategoryService.delete(
+        IdMap.getId("jeans")
+      )
+
+      expect(productCategoryRepository.softRemove).toBeCalledTimes(1)
```

Comment:
question: I am wondering if it make sense to softRemove a category? Is it to allow a user to recover it?

Reply:
No particular reason here. Just following the same methods as what other similar resources are doing (sales channels, product collection, etc.)

Reply:
My question is more about, should a category be soft deletable?

cc @olivermrbl